contracts/pcv/PCVGuardian.sol

```diff
@@ -76,114 +79,165 @@ contract PCVGuardian is IPCVGuardian, CoreRef {
     function withdrawETHToSafeAddress(
         address pcvDeposit,
         address payable safeAddress,
         uint256 amount,
-        bool pauseAfter,
         bool depositAfter
-    ) external override hasAnyOfThreeRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_SAFE_MOVER_ROLE, TribeRoles.GUARDIAN) {
-        require(isSafeAddress(safeAddress), "Provided address is not a safe address!");
-
-        pcvDeposit._ensureUnpaused();
-
+    ) external override beforeAndAfterWithdrawal(pcvDeposit, safeAddress, depositAfter) {
         IPCVDeposit(pcvDeposit).withdrawETH(safeAddress, amount);
+        emit PCVGuardianETHWithdrawal(pcvDeposit, safeAddress, amount);
+    }
 
-        if (pauseAfter) {
-            pcvDeposit._pause();
-        }
-
-        if (depositAfter) {
-            IPCVDeposit(safeAddress).deposit();
-        }
+    /// @notice withdraw funds from a pcv deposit, by calling the withdrawETH() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    /// @param safeAddress the destination address to withdraw to
+    /// @param basisPoints the percent in basis points [1-10000] if the deposit's balance to withdraw
+    /// @param depositAfter if true, attempts to deposit to the target PCV deposit
+    function withdrawETHRatioToSafeAddress(
+        address pcvDeposit,
+        address payable safeAddress,
+        uint256 basisPoints,
+        bool depositAfter
+    ) external override beforeAndAfterWithdrawal(pcvDeposit, safeAddress, depositAfter) {
+        require(basisPoints <= Constants.BASIS_POINTS_GRANULARITY, "PCVGuardian: basisPoints too high");
+        uint256 amount = (pcvDeposit.balance * basisPoints) / Constants.BASIS_POINTS_GRANULARITY;
+        require(amount != 0, "PCVGuardian: no value to withdraw");
 
+        IPCVDeposit(pcvDeposit).withdrawETH(safeAddress, amount);
         emit PCVGuardianETHWithdrawal(pcvDeposit, safeAddress, amount);
     }
 
-    /// @notice governor-or-guardian-only method to withdraw funds from a pcv deposit, by calling the withdraw() method on it
+    /// @notice withdraw funds from a pcv deposit, by calling the withdrawERC20() method on it
     /// @param pcvDeposit the deposit to pull funds from
     /// @param safeAddress the destination address to withdraw to
     /// @param amount the amount of funds to withdraw
-    /// @param pauseAfter whether to pause the pcv after withdrawing
     /// @param depositAfter if true, attempts to deposit to the target PCV deposit
     function withdrawERC20ToSafeAddress(
         address pcvDeposit,
         address safeAddress,
         address token,
         uint256 amount,
-        bool pauseAfter,
         bool depositAfter
-    ) external override hasAnyOfThreeRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_SAFE_MOVER_ROLE, TribeRoles.GUARDIAN) {
-        require(isSafeAddress(safeAddress), "Provided address is not a safe address!");
-
-        pcvDeposit._ensureUnpaused();
```

Comment:
Do we not need this to withdraw from paused deposits?

Reply:
we do, that's why before withdrawals the modifier reads the paused state & unpauses if the deposit is paused
![image](https://user-images.githubusercontent.com/6916865/178699200-cfc65d8c-165f-47fd-8a6a-79ee200fb181.png)